### PR TITLE
docs: add smart-release changelog

### PR DESCRIPTION
**Pull request by sdbondi**
Replaces changelog with one generated by [`cargo-smart-release`](https://crates.io/crates/cargo-smart-release)

`cargo smart-release --execute --bump auto`

Ref https://github.com/tari-project/tari-crypto/pull/88#issuecomment-1069158908

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+   
+   - Minor optimisation for lexicographical public key ordering: using reference to the underlying compressed array rather than copying the array onto the stack and then creating a reference
+
+### Commit Statistics
+
+<csr-read-only-do-not-edit/>
+
+ - 3 commits contributed to the release.
+ - 1 day passed between releases.
+ - 3 commits where understood as [conventional](https://www.conventionalcommits.org).
```
Comment (delta1) [inline CHANGELOG.md:37]:
```suggestion
 - 3 commits were understood as [conventional](https://www.conventionalcommits.org).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+
+ - <csr-id-09cc52787272ced3a1a8c9f2edc1e0221f9d8faa/> update rust toolchain (proc_macro_is_available feature)
+
+### Commit Statistics
+
+<csr-read-only-do-not-edit/>
+
+ - 9 commits contributed to the release over the course of 119 calendar days.
+ - 134 days passed between releases.
+ - 5 commits where understood as [conventional](https://www.conventionalcommits.org).
```
Comment (delta1) [inline CHANGELOG.md:73]:
```suggestion
 - 5 commits were understood as [conventional](https://www.conventionalcommits.org).
```

looks like it's wrong in the generator?

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+ - 7 unique issues were worked on: [#58](https://github.comgit//sdbondi/tari-crypto/issues/58), [#59](https://github.comgit//sdbondi/tari-crypto/issues/59), [#60](https://github.comgit//sdbondi/tari-crypto/issues/60), [#61](https://github.comgit//sdbondi/tari-crypto/issues/61), [#62](https://github.comgit//sdbondi/tari-crypto/issues/62), [#68](https://github.comgit//sdbondi/tari-crypto/issues/68), [#74](https://github.comgit//sdbondi/tari-crypto/issues/74)
```
Comment (delta1) [inline CHANGELOG.md:74]: I'd be surprised if these links worked?

Comment (sdbondi) [reply CHANGELOG.md:74]: yeah I think the owner of the repo would have to run this

**Comment by sdbondi** [comment]
Closing as this can be done by someone with write access

**Pull request closed without merging**